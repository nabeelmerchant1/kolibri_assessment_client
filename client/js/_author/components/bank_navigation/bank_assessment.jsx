import React                  from 'react';
import _                      from 'lodash';
import { connect }            from 'react-redux';

import appHistory             from '../../history';
import ListItem               from './list_item';
import * as AssessmentActions from '../../../actions/qbank/assessments';
import PublishButton          from './buttons/publish_button';
import EmbedButton            from './buttons/embed_button';
import EditButton             from './buttons/edit_button';
import DeleteButton           from './buttons/delete_button';
import PreviewButton          from './buttons/preview_button';
import DotLoader              from '../common/dot_loader';


function select() {
  return {};
}

export class BankAssessment extends React.Component {
  static propTypes = {
    assessment: React.PropTypes.shape({
      isPublished: React.PropTypes.bool.isRequired,
      bankId: React.PropTypes.string.isRequired,
      id: React.PropTypes.string.isRequired,
    }).isRequired,
    togglePublishAssessment: React.PropTypes.func.isRequired,
    onFocus: React.PropTypes.func.isRequired,
    deleteAssessment: React.PropTypes.func.isRequired,
  };

  constructor() {
    super();
    this.state = {
      deleting: false,
    };
  }

  deleteAssessment(...args) {
    if (confirm('Are you sure you want to delete this assessment?')) {
      this.setState({ deleting: true });
      this.props.deleteAssessment(...args);
    }
  }

  selectItem() {
    const { assessment } = this.props;
    if (assessment.isPublished) {
      appHistory.push(`banks/${assessment.bankId}/assessments/${assessment.id}/preview`);
      return;
    }
    appHistory.push(`banks/${assessment.bankId}/assessments/${assessment.id}`);
  }

  render() {
    const { assessment } = this.props;
    const displayName = _.get(assessment, 'displayName.text');
    // we pass empty functions to list item so that you cant open an assessment while its
    // being deleted.
    if (this.state.deleting) {
      return (
        <ListItem
          {...this.props}
          selectItem={() => {}}
          onFocus={() => {}}
        >
          <td className="au-deleting-assessment">
            <DotLoader />
          </td>
        </ListItem>
      );
    }
    return (
      <ListItem
        {...this.props}
        bank={assessment}
        selectItem={() => this.selectItem()}
        onFocus={this.props.onFocus}
      >
        <td>
          <i className="material-icons">description</i>
        </td>
        <td>{displayName}</td>
        <td>
          <PublishButton
            assessment={assessment}
            togglePublishAssessment={
              selectedAssessment => this.props.togglePublishAssessment(selectedAssessment)
            }
            onFocus={this.props.onFocus}
          />
        </td>
        <td>
          <div className="au-c-table__icons">
            <EmbedButton {...this.props} />
            <EditButton {...this.props} />
            <PreviewButton {...this.props} />
            <DeleteButton
              {...this.props}
              deleteAssessment={(...args) => this.deleteAssessment(...args)}
            />
          </div>
        </td>
      </ListItem>
    );
  }
}

export default connect(select, {
  ...AssessmentActions,
})(BankAssessment);
