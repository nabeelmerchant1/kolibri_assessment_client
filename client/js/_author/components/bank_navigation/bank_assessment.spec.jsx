import React                from 'react';
import { shallow }          from 'enzyme';
import { BankAssessment }   from './bank_assessment';

describe('bank_assessment component', () => {
  let props;
  let result;

  beforeEach(() => {
    props = {
      assessment: {
        isPublished: false,
        bankId: '',
        id: '',
        displayName: {
          text: 'Spec Name',
        },
      },
      togglePublishAssessment: () => {},
      onFocus: () => {},
      getEmbedCode: () => {},
      baseEmbedUrl: '',
      focused: false,
      deleteAssessment: () => {},
    };
    result = shallow(<BankAssessment {...props} />);
  });

  it('renders the component as a snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
