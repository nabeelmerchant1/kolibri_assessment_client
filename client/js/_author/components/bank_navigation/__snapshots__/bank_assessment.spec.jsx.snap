// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bank_assessment component renders the component as a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ListItem
    assessment={
        Object {
            "bankId": "",
            "displayName": Object {
              "text": "Spec Name",
            },
            "id": "",
            "isPublished": false,
          }
    }
    bank={
        Object {
            "bankId": "",
            "displayName": Object {
              "text": "Spec Name",
            },
            "id": "",
            "isPublished": false,
          }
    }
    baseEmbedUrl=""
    deleteAssessment={[Function]}
    focused={false}
    getEmbedCode={[Function]}
    onFocus={[Function]}
    selectItem={[Function]}
    togglePublishAssessment={[Function]}
>
    <td>
        <i
            className="material-icons"
        >
            description
        </i>
    </td>
    <td>
        Spec Name
    </td>
    <td>
        <PublishButton
            assessment={
                Object {
                    "bankId": "",
                    "displayName": Object {
                      "text": "Spec Name",
                    },
                    "id": "",
                    "isPublished": false,
                  }
            }
            onFocus={[Function]}
            togglePublishAssessment={[Function]}
        />
    </td>
    <td>
        <div
            className="au-c-table__icons"
        >
            <EmbedButton
                assessment={
                    Object {
                        "bankId": "",
                        "displayName": Object {
                          "text": "Spec Name",
                        },
                        "id": "",
                        "isPublished": false,
                      }
                }
                baseEmbedUrl=""
                deleteAssessment={[Function]}
                focused={false}
                getEmbedCode={[Function]}
                onFocus={[Function]}
                togglePublishAssessment={[Function]}
            />
            <EditButton
                assessment={
                    Object {
                        "bankId": "",
                        "displayName": Object {
                          "text": "Spec Name",
                        },
                        "id": "",
                        "isPublished": false,
                      }
                }
                baseEmbedUrl=""
                deleteAssessment={[Function]}
                focused={false}
                getEmbedCode={[Function]}
                onFocus={[Function]}
                togglePublishAssessment={[Function]}
            />
            <PreviewButton
                assessment={
                    Object {
                        "bankId": "",
                        "displayName": Object {
                          "text": "Spec Name",
                        },
                        "id": "",
                        "isPublished": false,
                      }
                }
                baseEmbedUrl=""
                deleteAssessment={[Function]}
                focused={false}
                getEmbedCode={[Function]}
                onFocus={[Function]}
                togglePublishAssessment={[Function]}
            />
            <DeleteButton
                assessment={
                    Object {
                        "bankId": "",
                        "displayName": Object {
                          "text": "Spec Name",
                        },
                        "id": "",
                        "isPublished": false,
                      }
                }
                baseEmbedUrl=""
                deleteAssessment={[Function]}
                focused={false}
                getEmbedCode={[Function]}
                onFocus={[Function]}
                togglePublishAssessment={[Function]}
            />
        </div>
    </td>
</ListItem>,
  "nodes": Array [
    <ListItem
      assessment={
            Object {
                  "bankId": "",
                  "displayName": Object {
                    "text": "Spec Name",
                  },
                  "id": "",
                  "isPublished": false,
                }
      }
      bank={
            Object {
                  "bankId": "",
                  "displayName": Object {
                    "text": "Spec Name",
                  },
                  "id": "",
                  "isPublished": false,
                }
      }
      baseEmbedUrl=""
      deleteAssessment={[Function]}
      focused={false}
      getEmbedCode={[Function]}
      onFocus={[Function]}
      selectItem={[Function]}
      togglePublishAssessment={[Function]}
>
      <td>
            <i
                  className="material-icons"
            >
                  description
            </i>
      </td>
      <td>
            Spec Name
      </td>
      <td>
            <PublishButton
                  assessment={
                        Object {
                              "bankId": "",
                              "displayName": Object {
                                "text": "Spec Name",
                              },
                              "id": "",
                              "isPublished": false,
                            }
                  }
                  onFocus={[Function]}
                  togglePublishAssessment={[Function]}
            />
      </td>
      <td>
            <div
                  className="au-c-table__icons"
            >
                  <EmbedButton
                        assessment={
                              Object {
                                    "bankId": "",
                                    "displayName": Object {
                                      "text": "Spec Name",
                                    },
                                    "id": "",
                                    "isPublished": false,
                                  }
                        }
                        baseEmbedUrl=""
                        deleteAssessment={[Function]}
                        focused={false}
                        getEmbedCode={[Function]}
                        onFocus={[Function]}
                        togglePublishAssessment={[Function]}
                  />
                  <EditButton
                        assessment={
                              Object {
                                    "bankId": "",
                                    "displayName": Object {
                                      "text": "Spec Name",
                                    },
                                    "id": "",
                                    "isPublished": false,
                                  }
                        }
                        baseEmbedUrl=""
                        deleteAssessment={[Function]}
                        focused={false}
                        getEmbedCode={[Function]}
                        onFocus={[Function]}
                        togglePublishAssessment={[Function]}
                  />
                  <PreviewButton
                        assessment={
                              Object {
                                    "bankId": "",
                                    "displayName": Object {
                                      "text": "Spec Name",
                                    },
                                    "id": "",
                                    "isPublished": false,
                                  }
                        }
                        baseEmbedUrl=""
                        deleteAssessment={[Function]}
                        focused={false}
                        getEmbedCode={[Function]}
                        onFocus={[Function]}
                        togglePublishAssessment={[Function]}
                  />
                  <DeleteButton
                        assessment={
                              Object {
                                    "bankId": "",
                                    "displayName": Object {
                                      "text": "Spec Name",
                                    },
                                    "id": "",
                                    "isPublished": false,
                                  }
                        }
                        baseEmbedUrl=""
                        deleteAssessment={[Function]}
                        focused={false}
                        getEmbedCode={[Function]}
                        onFocus={[Function]}
                        togglePublishAssessment={[Function]}
                  />
            </div>
      </td>
</ListItem>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <BankAssessment
        assessment={
                Object {
                        "bankId": "",
                        "displayName": Object {
                          "text": "Spec Name",
                        },
                        "id": "",
                        "isPublished": false,
                      }
        }
        baseEmbedUrl=""
        deleteAssessment={[Function]}
        focused={false}
        getEmbedCode={[Function]}
        onFocus={[Function]}
        togglePublishAssessment={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": BankAssessment {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "assessment": Object {
            "bankId": "",
            "displayName": Object {
              "text": "Spec Name",
            },
            "id": "",
            "isPublished": false,
          },
          "baseEmbedUrl": "",
          "deleteAssessment": [Function],
          "focused": false,
          "getEmbedCode": [Function],
          "onFocus": [Function],
          "togglePublishAssessment": [Function],
        },
        "refs": Object {},
        "state": Object {
          "deleting": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ListItem
          assessment={
                    Object {
                              "bankId": "",
                              "displayName": Object {
                                "text": "Spec Name",
                              },
                              "id": "",
                              "isPublished": false,
                            }
          }
          bank={
                    Object {
                              "bankId": "",
                              "displayName": Object {
                                "text": "Spec Name",
                              },
                              "id": "",
                              "isPublished": false,
                            }
          }
          baseEmbedUrl=""
          deleteAssessment={[Function]}
          focused={false}
          getEmbedCode={[Function]}
          onFocus={[Function]}
          selectItem={[Function]}
          togglePublishAssessment={[Function]}
>
          <td>
                    <i
                              className="material-icons"
                    >
                              description
                    </i>
          </td>
          <td>
                    Spec Name
          </td>
          <td>
                    <PublishButton
                              assessment={
                                        Object {
                                                  "bankId": "",
                                                  "displayName": Object {
                                                    "text": "Spec Name",
                                                  },
                                                  "id": "",
                                                  "isPublished": false,
                                                }
                              }
                              onFocus={[Function]}
                              togglePublishAssessment={[Function]}
                    />
          </td>
          <td>
                    <div
                              className="au-c-table__icons"
                    >
                              <EmbedButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                              <EditButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                              <PreviewButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                              <DeleteButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                    </div>
          </td>
</ListItem>,
        "_debugID": 2,
        "_renderedOutput": <ListItem
          assessment={
                    Object {
                              "bankId": "",
                              "displayName": Object {
                                "text": "Spec Name",
                              },
                              "id": "",
                              "isPublished": false,
                            }
          }
          bank={
                    Object {
                              "bankId": "",
                              "displayName": Object {
                                "text": "Spec Name",
                              },
                              "id": "",
                              "isPublished": false,
                            }
          }
          baseEmbedUrl=""
          deleteAssessment={[Function]}
          focused={false}
          getEmbedCode={[Function]}
          onFocus={[Function]}
          selectItem={[Function]}
          togglePublishAssessment={[Function]}
>
          <td>
                    <i
                              className="material-icons"
                    >
                              description
                    </i>
          </td>
          <td>
                    Spec Name
          </td>
          <td>
                    <PublishButton
                              assessment={
                                        Object {
                                                  "bankId": "",
                                                  "displayName": Object {
                                                    "text": "Spec Name",
                                                  },
                                                  "id": "",
                                                  "isPublished": false,
                                                }
                              }
                              onFocus={[Function]}
                              togglePublishAssessment={[Function]}
                    />
          </td>
          <td>
                    <div
                              className="au-c-table__icons"
                    >
                              <EmbedButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                              <EditButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                              <PreviewButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                              <DeleteButton
                                        assessment={
                                                  Object {
                                                            "bankId": "",
                                                            "displayName": Object {
                                                              "text": "Spec Name",
                                                            },
                                                            "id": "",
                                                            "isPublished": false,
                                                          }
                                        }
                                        baseEmbedUrl=""
                                        deleteAssessment={[Function]}
                                        focused={false}
                                        getEmbedCode={[Function]}
                                        onFocus={[Function]}
                                        togglePublishAssessment={[Function]}
                              />
                    </div>
          </td>
</ListItem>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BankAssessment
    assessment={
        Object {
            "bankId": "",
            "displayName": Object {
              "text": "Spec Name",
            },
            "id": "",
            "isPublished": false,
          }
    }
    baseEmbedUrl=""
    deleteAssessment={[Function]}
    focused={false}
    getEmbedCode={[Function]}
    onFocus={[Function]}
    togglePublishAssessment={[Function]}
/>,
}
`;
